### DISCUSSION TITLE
tac_plus authentication with django-auth-tacacs config param not found

### DISCUSSION THREAD

**Discussion by andyb2000** [netbox-community/netbox#12365] [Help Wanted!] [answered]
Hi all,
I'm trying to use https://pypi.org/project/django-auth-tacacs/ to do tacacs authentication for my netbox installation but I'm hitting what is probably a very silly issue!
I've added it, and added configuration to my /opt/netbox/netbox/netbox/configuration.py:
`REMOTE_AUTH_BACKEND = 'django_auth_tacacs.django.TACACSPlusAuthenticationBackend'

TACACSPLUS_HOST = 'x.x.x.x'

TACACSPLUS_PORT = 49

TACACSPLUS_SECRET = 'mysecret'

TACACSPLUS_SESSION_TIMEOUT = 10

TACACSPLUS_AUTH_PROTOCOL = 'ascii'

TACACSPLUS_AUTOCREATE_USERS = True
`

But when trying to login I get the error thrown on the webui:
<class 'AttributeError'>

'Settings' object has no attribute 'TACACSPLUS_HOST'

Python version: 3.10.6
NetBox version: 3.4.6

So it's not being able to read the TACACSPLUS_HOST value.
I've checked the code to django-auth-tacacs and it reads using standard django:
  from django.conf import settings
<snip>
  settings.TACACSPLUS_HOST

But those don't seem to be passing through. Am I missing something obvious here?

Thank you!

**Comment by candlerb** [comment, marked as answer]
Presumably you've put them in `configuration.py`, not `settings.py`?

My guess is it might be necessary to hack Netbox's `settings.py` to get this to work. You're not supposed to do that, but if it works, you could feed back a PR to make this a supported configuration.

**Comment by andyb2000** [comment]
Aha! So adding the configuration into settings.py has solved that.
Yes it's not ideal but it works, I'll write a quick PR to bring the settings in correctly.

**Reply by Dark-Helmet1** [reply]
Has the PR been approved?
I just tried this and had to put the settings into settings.py.

After installing  the auth module via PIP you will need to edit the source code of site-packages/django_auth_tacacs/netbox.py to make it pick up the new user settings.  

Comment out:
from django.contrib.auth.models import User

Add:
from django.contrib.auth import get_user_model
User = get_user_model()
